### PR TITLE
chore(query): refactor atomWithQuery

### PR DESCRIPTION
**Pull request by dai-shi**
Thanks to #431 and #444, we can simplify the async atom.

### CODE REVIEW

File: tests/query/atomWithQuery.test.tsx [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-import { Provider as ProviderOrig, atom, useAtom } from '../../src/'
+import { atom, useAtom } from '../../src/'
 import fakeFetch from './fakeFetch'
 import { atomWithQuery } from '../../src/query'
+import { getTestProvider } from '../testUtils'
 
-// FIXME Provider-less mode fails
-// const Provider = getTestProvider()
-const Provider = ProviderOrig
+const Provider = getTestProvider()
```
Comment (dai-shi) [inline tests/query/atomWithQuery.test.tsx:8]: Fixed.

File: src/query/atomWithQuery.ts [hunk 2]
```diff
@@ -55,19 +55,15 @@ export function atomWithQuery<
...
-            if (pending.fulfilled) {
-              set(pendingAtom, createPending<TData>()) // new fetch
-            }
+            set(
+              dataAtom,
+              new Promise<TData>(() => {}) // new fetch
+            )
             action.initializer(getQueryClient(get, set))
           } else if (action.type === 'data') {
             const data = get(dataAtom)
```
Comment (Thisen) [inline src/query/atomWithQuery.ts:64]:
```suggestion
            const data: TData = get(dataAtom)
```

File: src/query/atomWithQuery.ts [hunk 3]
```diff
@@ -55,19 +55,15 @@ export function atomWithQuery<
...
-            if (data === null || !equalityFn(data, action.data)) {
+            if (data === null || !equalityFn(data as TData, action.data)) {
```
Comment (Thisen) [inline src/query/atomWithQuery.ts:65]:
```suggestion
            if (data === null || !equalityFn(data, action.data)) {
```

**Review by Thisen — APPROVED** [review]
Awesome! Got a suggestion to a type.

**Review by Aslemammad — COMMENTED** [review]
Looks great to me; I think @Thisen suggestions are great to commit.

**Comment by dai-shi** [comment]
Thanks for the suggestion. But it failed...
I've got proper typing. This might be too complicated, but will see.

**Pull request merged**